src/format/FormatChecker.c

```diff
@@ -0,0 +1,167 @@
+//
+// Copyright (c) 2019 Takpit. All rights reserved.
+//
+
+#include "proto_nmea/format/FormatChecker.h"
+
+//region Helper Functions Declaration
+
+static int8_t validateInitialMessageFormat(const char *rawMessage);
+
+static int8_t validateMessageLengthInRange(size_t messageSize);
+
+static int8_t validateChecksumFormat(const char *message, size_t messageSize, size_t *checksumIdealIndex);
+
+static int8_t validateMessageType(const char *message);
+
+static int8_t validateChecksumData(const char *message, size_t checksumDelimiterIndex);
+
+//endregion
+
+//region API
+
+int8_t validateMessageFormat(const char *message)
+{
+    size_t messageSize = strlen(message);
+
+    return validateFixedSizeMessageFormat(message, messageSize);
+}
+
+int8_t validateFixedSizeMessageFormat(const char *rawMessage, size_t messageSize)
+{
+    int8_t errorCode = validateInitialMessageFormat(rawMessage);
+    if (errorCode)
+        return errorCode;
+
+    // Validate message ending fits protocol
+    if (rawMessage[messageSize - 2] != PROTOCOL_STOP_CHAR_1 ||
```

Comment:
Use constants (defines) for offsets & length instead of explicit values

Reply:
In this case we're merely checking the 2 last elements of the array, I don't there's a need to name it somehow - Care to tell me if I'm wrong?